core/block.js

```diff
@@ -1445,3 +1445,19 @@ Blockly.Block.prototype.allInputsFilled = function(opt_shadowBlocksAreFilled) {
 
   return true;
 };
+
+/**
+ * @return {string} A description of this block using type name and id.
+ */
+Blockly.Block.prototype.describe = function() {
```

Comment:
Why not make this a toString() function?  Then you don't need to call it explicitly.